bears/r/FormatRBear.py

```diff
@@ -1,24 +1,22 @@
-from coalib.bearlib.abstractions.Lint import Lint
-from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.abstractions.Linter import linter
 
 
-class FormatRBear(Lint, LocalBear):
-    executable = "Rscript"
-    arguments = ("-e 'library(formatR)' "
-                 "-e 'formatR::tidy_source(\"{filename}\")'")
-    prerequisite_command = ['Rscript', '-e', "'library(formatR)'"]
-    prerequisite_fail_msg = "Please install formatR for this bear to work."
-    diff_message = "Formatting can be improved."
-    gives_corrected = True
+@linter(executable='Rscript',
+        output_format='corrected',
+        prerequisite_check_command=('Rscript', '-e',
+                                    "library(formatR)"),
+        prerequisite_check_fail_message='Please install formatR for this bear '
+                                        'to work.')
+class FormatRBear:
+    """
+    This bear checks and corrects formatting of R Code using known formatR
+    utility.
+    """
     LANGUAGES = "R"
 
-    def process_output(self, output, filename, file):
-        output = output[:-1] + (output[-1].strip()+"\n",)
-        return Lint.process_output(self, output, filename, file)
+    @staticmethod
+    def create_arguments(filename, file, config_file):
 
-    def run(self, filename, file):
-        '''
-        This bear checks and corrects formatting of R Code using
-        known formatR utility.
-        '''
-        return self.lint(filename, file)
+        rcode = ("formatR::tidy_source(source =\""+filename+"\"")
+        args = ('-e', "library(formatR)", '-e',)
+        return args + (rcode,)
```

Comment:
does my suggestion work with
```
rcode = ('library(formatR)\n'
         'formatR::tidy_source(source="' + filename + '")'
```
\+ I'm just seeing -> you forgot a closing brace ) inside the tidy_source command, interestingly it seemed to work anyway.

\+ please escape special characters for the filename, you can't use everything in R :+1: